packages/react-form-state/README.md

```diff
@@ -0,0 +1,947 @@
+# `@shopify/react-form-state`
+
+[![Build Status](https://travis-ci.org/Shopify/quilt.svg?branch=master)](https://travis-ci.org/Shopify/quilt)
+
+[![License: MIT](https://img.shields.io/badge/License-MIT-green.svg)](LICENSE.md) [![npm version](https://badge.fury.io/js/%40shopify%2Freact-formwell.svg)](https://badge.fury.io/js/%40shopify%2Freact-formwell.svg) [![npm bundle size (minified + gzip)](https://img.shields.io/bundlephobia/minzip/@shopify/react-formwell.svg)](https://img.shields.io/bundlephobia/minzip/@shopify/react-formwell.svg)
+
+Manage react forms tersely and type-safe with no magic.
+
+## Another form library?
+
+Good forms in React are surprisingly tough. A good api needs to support complex usecases with large numbers of fields split up across multiple components and explicit validation. It also needs to be quick to get started with and work for the simplest of forms cleanly. In meeting these goals, a package needs to balance developer experience, flexibility, and performance.
+
+Several popular community packages sit at different places on the spectrum for these requirements, and all are solid choices for many projects.
+
+**Examples**
+
+- [formik](https://github.com/jaredpalmer/formik) offers a terse but relatively unmagical way of managing form state with minimal dependencies.
+- [redux-form](https://redux-form.com/7.4.2/) offers a redux integrated solution that can be quite magical and give you the power of the redux devtools for debugging broken states.
+
+At Shopify we wanted our form solution to be true to the Web Foundations team's core values. We wanted it to be **explicit**, **type-safe**, and to **prioritize user experience** while **keeping code quality and developer happiness high**.
```

Comment:
should we link these to our web foundations repo? In particular:

1. Should we explicitly address https://github.com/Shopify/web-foundation/blob/master/Principles/5%20-%20Community%20over%20ownership.md and why the advantages of writing our own forms library outweigh the costs?
2. Should we link to https://github.com/Shopify/web-foundation/blob/master/Principles/3%20-%20Explicit%20over%20automatic.md ?
3. See my suggestion via https://github.com/Shopify/web-foundation/issues/32 to add docs about our preferences for type safety.